### PR TITLE
[SPARK-50029][SQL] Make `StaticInvoke` compatible with the method that return `Any`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to make `StaticInvoke` compatible with the method that return `Any`.

### Why are the changes needed?
Currently, our `StaticInvoke` does not support calling the method with a return type signature of `Any`(actually, the type of return value may be `different data type`), while `Invoke` supports it, let's align it.

### Does this PR introduce _any_ user-facing change?
No, only for spark developer.

### How was this patch tested?
- Add new UT.
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala [hunk 1]
```diff
@@ -755,6 +755,31 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     val genCode = StaticInvoke(TestFun.getClass, IntegerType, "foo", arguments).genCode(ctx)
     assert(!genCode.code.toString.contains("boxedResult"))
   }
+
+  test("StaticInvoke call return `any` method") {
+    val cls = TestStaticInvokeReturnAny.getClass
+    Seq((0, IntegerType, true), (1, IntegerType, true), (2, IntegerType, false)).foreach {
```
Comment (panbingkun) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala:761]: - case 0 - `(0, IntegerType, true)` will hit line `325` of `objects.scala`
- case 1 - `(1, IntegerType, true)` will hit line `332` of `objects.scala`
- case 2 - `(2, IntegerType, true)` will hit line `340` of `objects.scala`

**Comment by panbingkun** [comment]
cc @cloud-fan

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala:761]: This UT can hit three points where `cast` is added.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**